### ISSUE TITLE
CockroachDB column families causes invalid models.go

### ISSUE THREAD

**Issue by angaz** [sqlc-dev/sqlc#1556]
### Version

1.12.0

### What happened?

There is a CockroachDB feature where you can group columns to be saved in a separate row: https://www.cockroachlabs.com/docs/v21.2/column-families.html

When adding this to my migrations to see if it would work, I realized that there was a bug. I got multiple `Family` keys in the generated model struct. Example in the Playground link.

I don't know how `sqlc` should react to this case because it technically is not a valid Postgres `CREATE TABLE` command.

Just thought that I would report the bug for completeness sake.

### Database schema

```sql
CREATE TABLE test (
    id INT PRIMARY KEY,
    last_accessed TIMESTAMP,
    data BYTES,
    FAMILY f1 (id, last_accessed),
    FAMILY f2 (data)
);
```

### SQL queries

```
-- name: Test :many
SELECT * FROM test;
```

### Configuration

```json
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "postgresql",
      "schema": "query.sql",
      "queries": "query.sql"
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/c78c86646951d3d8c4bf1e40f0bc9235d7fe867015b1a80c9473cec3ce0c8be2

### What operating system are you using?

Linux

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by kyleconroy** [comment]
The PostgreSQL parser sees this as a valid create table statement since it doesn't understand the `FAMILY` syntax. Adding this as another issue dependent on #1389.

**Comment by SN9NV** [comment]
I think I badly communicated the problem I was trying to report. The resulting generated Go code is not valid because the same `Family` key is duplicated in the model struct.

```
type Test struct {
	ID           int32
	LastAccessed sql.NullTime
	Data         interface{}
	Family       interface{}
	Family       interface{}
}
```

Would you be interested in me having a look if I could add the cockroach parser as an engine? Have you had a look in the past? Is this a viable option? I see their versioning doesn't lend well to being used as a Go package, so I would have to extract it as a package and maybe ask them to extract it upstream if it works for us here. What do you think?

**Comment by kyleconroy** [comment]
Adding the CockroachDB parser would be the first step in adding the engine. There are some large changes that we're going to need to make to support Cockroach. Closing this issue for now as it won't work until official support lands.

**Comment by ppaulweber** [comment]
@kyleconroy are there any updates on this issue regarding a possible CRDB parser integration and golang code generation of CRDB SQL?
